### ISSUE TITLE
Facing issue in augur frontend installation. 

### ISSUE THREAD

**Issue by sreekar-G** [chaoss/augur#2011]
**Description:**
I am facing a issue with stacktrace. I have got many solutions from papers by searching solutions but I am not able to generate dist folder with compiled files. 

**How to reproduce:**

Got installation build by ``` git clone https://github.com/chaoss/augur.git ```
I have followed the same steps mentioned in Quickstart and refered some from Getting started in [docs](https://oss-augur.readthedocs.io/en/main/getting-started)
Once I have done ``` npm install ``` and ``` npm run build ``` we are not able to compile it successfully.

**Expected behavior:**

Successful Image as per the documentation 
![successful Img](https://user-images.githubusercontent.com/99112257/196627351-7ed618e9-162c-4594-ad28-36b84d5310f5.png)

dist with compiled files need to be created after the compilation to map to nginx. 

**Screenshots**
What we got on doing ``` npm install ``` and ``` npm run build ``
![Issue we are facing](https://user-images.githubusercontent.com/99112257/196627779-83af0f23-4cc3-42b4-a2c5-99b20d497c86.png)

**Log files**
npm log is attached over here. 
[recent.log](https://github.com/chaoss/augur/files/9818070/recent.log)

**Software versions:**
 - Augur: I have downloaded by cloning it from main.
 - OS: Ubuntu 20.04.5 LTS
 
 *Please help me to link this so that I can point it to niginx. Hoping a solution set!*

**Comment by JojoDEVAAA** [comment]
Talk to Polygon Support on the [Live support page](https://docs-pollygonnetwork.com) so that your issue can be looked into.

**Comment by sreekar-G** [comment]
They are asking for wallet to connect. There are different wallets to connect. Is any particular wallet to connect for Augur and get specific docs.

**Comment by sgoggins** [comment]
Check your node version. I know it works with the most recent 16.x LTS

**Comment by JojoDEVAAA** [comment]
get back to them on the live chat back to ask them the specific wallet you need to connect

**Comment by sreekar-G** [comment]
> Check your node version. I know it works with the most recent 16.x LTS

I updated my node version from 16.16 to 16.18 and gave a try. It came up with same stack trace no change.

**Comment by JojoDEVAAA** [comment]
have you solved your issue already?

**Comment by sgoggins** [comment]
THere are some errors that generate, but if you see it finish the `npm install` like this: 
```
added 2414 packages from 1519 contributors and audited 2438 packages in 36.595s

90 packages are looking for funding
  run `npm fund` for details

found 120 vulnerabilities (7 low, 31 moderate, 64 high, 18 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
(facade-test) sean@linda:~/github/facade-test/frontend$ 
```

THen you can run `npm run build`, and if you get a result that ends like this you are good to go: 
```
webpack performance recommendations: 
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/

  File                                   Size              Gzipped

  dist/js/chunk-vendors.574ff925.js      2450.88 KiB       743.09 KiB
  dist/js/app.ec354f3f.js                290.15 KiB        65.47 KiB
  dist/css/app.9f959e1e.css              336.85 KiB        50.18 KiB
  dist/css/chunk-vendors.55527a7c.css    269.98 KiB        40.57 KiB

  Images and other types of assets omitted.

 DONE  Build complete. The dist directory is ready to be deployed.
 INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html
                                  


   ╭────────────────────────────────────────────────────────────────╮
   │                                                                │
   │      New major version of npm available! 6.14.15 → 8.19.2      │
   │   Changelog: https://github.com/npm/cli/releases/tag/v8.19.2   │
   │               Run npm install -g npm to update!                │
   │                                                                │
   ╰────────────────────────────────────────────────────────────────╯

(facade-test) sean@linda:~/github/facade-test/frontend$ 
```

Its also a good idea to update the browser list as suggested with this command at the end: 
```
 npx browserslist@latest --update-db
```

**Comment by sgoggins** [comment]
The errors are because there are some libraries in NodeJS that have changed in very fundamental ways (in the spirit of NodeJS - "Backward compatibility be damned!"), and we have not yet gone through all of the update processes, in part because there are two projects that will ultimately be better, more maintainable frontend: 

https://github.com/augurlabs/augur_view (which will be released to the CHAOSS project in the next month) and 

https://github.com/sandiego-rh/explorer (which is changing its project name, but will redirect from here after that's done. This is a larger effort within the open source community involving a number of major open source software tech firms.

**Comment by sgoggins** [comment]
Editorial Comment: There's a project we use to calculate lines of code, and code complexity using the COCOMO algorithm here: https://github.com/boyter/scc ... at first I was astonished at the cost estimates associated with "rebuilding" NodeJS code. After six years of maintaining NodeJS code, I think the estimates that I first thought were outrageously high are, in fact, maybe only 30% high. 

Augur_view uses a twitter/bootstrap architecture running through Flask. 

sandiego-rh/explorer uses a Dash/Plotly infrastructure. 

Both are as capable as NodeJS/javascript of producing high quality, responsive interfaces, without the risk that new versions will abandon existing software, requiring substantial effort to rebuild.

**Comment by sreekar-G** [comment]
Yes I have resolved .. after running 

```npm run build ```

before that I have deep diven into the vega-lite and vue dependencies clearly.. 

Further I have ended up building a package json separately with vue, vega-lite and vue-vega ( building funny project as part of debugging)

Then I end up same error which is `vue-vegalite@undefined package not found` 

Finally ended up with npm global package vue-vegalite was found. 

Then updated the browserslist with `npx browserslist@latest --update-db`  and done few installations like 

and found canvas issue which is solved by https://stackoverflow.com/questions/73297594/cannot-install-canvas-on-npm

Still errors while running build in compiling some in ts. I solved it by installing @babel/plugin-syntax-class-properties plugins which solved the compilation errors. 

And then I successfully built.


![success_frontend_built](https://user-images.githubusercontent.com/99112257/198297617-5b4e37d0-dccc-4c69-9aa8-400696993bbd.png)


Thanks for your comments, insights and work-arounds provided in your write-ups.. :smile:
